### DISCUSSION TITLE
How to execute routes without using the REST endpoints ?

### DISCUSSION THREAD

**Discussion by hhertoghe** [fastapi/fastapi#11831] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
/
```

### Description

Hello,
I would like to be able to import the fastapi app in one of my application and to call endpoints of routers without passing by the HTTP endpoints & overloading the infrastructure.
Today I'm using testclient for this in production but this seems not the best solution as I'm quickly reaching memory & cpu problems. 
Any idea what I could do here to "mock" rest calls ?

### Operating System

Linux

### FastAPI Version

0.95.2

### Pydantic Version

1.10.2

### Python Version

3.11.8

**Comment by n8sty** [comment]
I'm not really sure what you're trying to do here but here's a pattern that should work:

What I imagine your original is like
```
@app.get("/")
def some_get_route(...):
    # Do something in here
```

What I suggest:
```
@app.get("/"
def some_get_route(...):
    data = some_function(...)
    return data
```

This decouples your logic a bit and allows calling `some_function` anywhere you want. Otherwise, it's possible to simply just call functions decorated as routes as if they were functions so just call the function you want to use without wrapping it in the `TestClient`.

**Reply by hhertoghe** [reply]
I would like to have the same logic applied as if it's called by the HTTP endpoint (pydantic validation, ...)
How to call the functions decorated as routes ? You have to pass the expected parameters I guess ?
